daam/run/demo.py

```diff
@@ -0,0 +1,158 @@
+import math
+from threading import Lock
+from typing import Any, List
+
+import numpy as np
+from diffusers import StableDiffusionPipeline
+from matplotlib import pyplot as plt
+import gradio as gr
+import torch
+
+from daam import trace
+from daam.utils import set_seed, expand_image
+
+
+def get_tokenizing_mapping(prompt: str, tokenizer: Any) -> List[List[int]]:
+    tokens = tokenizer.tokenize(prompt)
+    merge_idxs = []
+    words = []
+    curr_idxs = []
+    curr_word = ''
+
+    for i, token in enumerate(tokens):
+        curr_idxs.append(i+1)  # because of the [CLS] token
+        curr_word += token
+        if '</w>' in token:
+            merge_idxs.append(curr_idxs)
+            curr_idxs = []
+            words.append(curr_word[:-4])
+            curr_word = ''
+
+    return merge_idxs, words
+
+
+def main():
+    # argparse?
+
+    model_id = "stabilityai/stable-diffusion-2-base"
+    device = "cuda"
+    pipe = StableDiffusionPipeline.from_pretrained(
+        model_id, use_auth_token=True,
+    ).to(device)
+    lock = Lock()
+
+    @torch.no_grad()
+    def predict(prompt, inf_steps, threshold):
+        merge_idxs, words = get_tokenizing_mapping(prompt, pipe.tokenizer)
+        with torch.cuda.amp.autocast(dtype=torch.float16), lock:
+            try:
+                plt.close('all')
+            except:
+                pass
+
+            gen = set_seed(0)
+            with trace(pipe) as tc:  # low memory?
```

Comment:
It's okay without low memory when we don't care about generating heat maps for each head.